app/helpers/gtl_helper.rb

```diff
@@ -0,0 +1,125 @@
+module GtlHelper
+  def gtl_selected_records
+    records = @edit[:object_ids] unless @edit.nil? || @edit[:object_ids].nil?
+    records = @edit[:pol_items] unless @edit.nil? || @edit[:pol_items].nil?
+    records = params[:rec_ids] unless params.nil? || params[:rec_ids].nil?
+    records = records.map(&:to_i) if !records.nil? && records.first.kind_of?(String)
+    records = records.map(&:id) if !records.nil? && !records.first.kind_of?(Integer)
+    records
+  end
+
+  def miq_tile_view
+    content_tag(
+      'miq-tile-view', '',
+      "ng-if"            => "dataCtrl.gtlType === 'grid' || dataCtrl.gtlType === 'tile'",
```

Comment:
just wondering, why does angular conditions appear inside ruby code?

Reply:
I am just splitting the partial into helpers. This logic comes from the partial. No changes here.

Reply:
yeah, overall major :-1: for lack of separation of concerns

Reply:
Well the helper here generates haml string which is angular's components and such, so this is OK. As far as I discussed this with Martin he wanted to generate whole page using partials so they can be much easily tested. I understand that it's somewhat harder to read HTML inside ruby which is essentially JS condition, but I wouldn't be too hard on this part of code, since it brings much more goodies with option to test this much thoroughly.